iothub_client/samples/iothub_client_device_cbor_sample/iothub_client_device_cbor_sample.c

```diff
@@ -0,0 +1,956 @@
+    memset(&iot_hub_car, 0, sizeof(Car));
+    parse_cbor_desired_property(update_state, payload, &iot_hub_car);
+
+    // Update device_car if twin desired properties received from IoT Hub are different than what it stored on device.
+    // change_oil_reminder
+    if (device_car.change_oil_reminder != iot_hub_car.change_oil_reminder)
+    {
+        bool temp_value = device_car.change_oil_reminder;
+        device_car.change_oil_reminder = iot_hub_car.change_oil_reminder;
+        (void)printf("The device client updated `%s` locally from %s to %s.\n", twin_property_change_oil_reminder_name, temp_value ? "true" : "false", device_car.change_oil_reminder ? "true" : "false");
+        update = true;
+    }
+
+    // state.max_speed
+    if (iot_hub_car.state.max_speed != 0)
+    {
+        if (device_car.state.max_speed != iot_hub_car.state.max_speed)
+        {
+            if (iot_hub_car.state.max_speed > 260)
+            {
+                (void)printf("Desired `max_speed` of %" PRIu64 " exceeds device capability. Rejecting update.\n", iot_hub_car.state.max_speed);
+            }
+            else
+            {
+                uint64_t temp_value = device_car.state.max_speed;
+                device_car.state.max_speed = iot_hub_car.state.max_speed;
+                (void)printf("The device client updated `%s` locally from %" PRIu64 " to %" PRIu64 ".\n", twin_property_state_max_speed_name, temp_value, iot_hub_car.state.max_speed);
+                update = true;
+            }
+        }
+    }
+    else
+    {
+        (void)printf("`max_speed` cannot be 0. Rejecting update.\n");
+    }
+
+    // state.software_version
+    if (iot_hub_car.state.software_version != 0)
+    {
+        if (device_car.state.software_version != iot_hub_car.state.software_version)
+        {
+            double temp_value = device_car.state.software_version;
+            device_car.state.software_version = iot_hub_car.state.software_version;
+            (void)printf("The device client updated `%s` locally from %.*f to %.*f.\n", twin_property_state_software_version_name, 1, temp_value, 1, iot_hub_car.state.software_version);
+            update = true;
+        }
+    }
+    else
+    {
+        (void)printf("`software_version` cannot be 0.0. Rejecting update.\n");
+    }
+
+    // If any property updates, send reported properties to IoT Hub.
+    if (update)
+    {
+        (void)printf("Reporting properties to IoT Hub.\n");
+        build_cbor_reported_property(reported_property_buffer, REPORTED_PROPERTY_BUFFER_SIZE, &reported_property_length);
+        int rc = IoTHubDeviceClient_SendReportedState(iothub_client_handle, reported_property_buffer, reported_property_length, twin_reported_properties_callback, NULL);
+        if (rc != IOTHUB_CLIENT_OK)
+        {
+            (void)printf("Failed to send a reported property PATCH request to the IoT Hub: return code %d.\n", rc);
+            exit(rc);
+        }
+    }
+
+    (void)printf("\n"); // Formatting
```

Comment:
a lot of this logic seems to be copied from the `get_twin_async_callback`. please merge them in a static function called by each to reduce code size.

Reply:
They are very similar, but slightly different enough, (and not enough code) that I think it is easiest to digest as is.